File: 백준/Unrated/33667. 루미의 28번째 생일/루미의 28번째 생일.py
from fractions import Fraction
 
MINUTE_SECONDS = 60
HOUR_SECONDS = 60 * MINUTE_SECONDS
DAY_SECONDS = 24 * HOUR_SECONDS
WEEK_SECONDS = 7 * DAY_SECONDS

 
def is_leap_year(year):
    return (year % 4 == 0 and year % 100 != 0) or (year % 400 == 0)


def days_in_month(year, month):
    if is_leap_year(year) and month == 2:
        return 29
    month_table = [0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31]
    return month_table[month]


def to_seconds(y, m, d, h, minute, s):
    total_seconds = 0
    leap_year_count = (y - 1) // 4 - (y - 1) // 100 + (y - 1) // 400
    total_seconds += (365 * (y - 1) + leap_year_count) * DAY_SECONDS
    for month in range(1, m):
        total_seconds += days_in_month(y, month) * DAY_SECONDS
    total_seconds += (d - 1) * DAY_SECONDS
    total_seconds += h * HOUR_SECONDS
    total_seconds += minute * MINUTE_SECONDS
    total_seconds += s
    return total_seconds


def fraction_in_month(y, m, d, h, minute, s):
    total_seconds = 0
    total_seconds += (d - 1) * DAY_SECONDS
    total_seconds += h * HOUR_SECONDS
    total_seconds += minute * MINUTE_SECONDS
    total_seconds += s
    return Fraction(total_seconds, days_in_month(y, m) * DAY_SECONDS)


def fraction_in_year(y, m, d, h, minute, s):
    total_seconds = 0
    for month in range(1, m):
        total_seconds += days_in_month(y, month) * DAY_SECONDS
    total_seconds += (d - 1) * DAY_SECONDS
    total_seconds += h * HOUR_SECONDS
    total_seconds += minute * MINUTE_SECONDS
    total_seconds += s
    return Fraction(total_seconds, (365 + is_leap_year(y)) * DAY_SECONDS)


def sol():
    a1, a2, a3, a4, a5, a6 = map(int, input().split())
    b1, b2, b3, b4, b5, b6 = map(int, input().split())
    unit = input()

    birth_total_seconds = to_seconds(a1, a2, a3, a4, a5, a6)
    cur_total_seconds = to_seconds(b1, b2, b3, b4, b5, b6)

    diff_seconds = cur_total_seconds - birth_total_seconds

    if unit == "Year":
        current_fraction = fraction_in_year(b1, b2, b3, b4, b5, b6)
        birth_fraction = fraction_in_year(a1, a2, a3, a4, a5, a6)
        is_full_year = 1 if current_fraction >= birth_fraction else 0
        print(b1 - a1 - 1 + is_full_year)
    elif unit == "Month":
        current_fraction = fraction_in_month(b1, b2, b3, b4, b5, b6)
        birth_fraction = fraction_in_month(a1, a2, a3, a4, a5, a6)
        is_full_month = 1 if current_fraction >= birth_fraction else 0
        print((b1 - a1) * 12 + b2 - a2 - 1 + is_full_month)
    elif unit == "Day":
        print(diff_seconds // DAY_SECONDS)


T = int(input())
for _ in range(T):
    sol()

File: 백준/Gold/33667. 루미의 28번째 생일/루미의 28번째 생일.cs
using System;

namespace Baekjoon
{
    public class CustomDateTime
    {
        public Int32 year { get; set; }
        public Int32 month { get; set; }
        public Int32 day { get; set; }
        public Int32 hour { get; set; }
        public Int32 minute { get; set; }
        public Int32 second { get; set; }


        public CustomDateTime(Int32 year, Int32 month, Int32 day,
            Int32 hour, Int32 minute, Int32 second)
        {
            this.year = year;
            this.month = month;
            this.day = day;
            this.hour = hour;
            this.minute = minute;
            this.second = second;
        }
    }

    public class Fraction
    {
        public Int64 numerator { get; set; }
        public Int64 denominator { get; set; }

        public Fraction(Int64 numerator, Int64 denominator)
        {
            this.numerator = numerator;
            this.denominator = denominator;
        }

        public static Boolean operator >=(Fraction f1, Fraction f2)
        {
            return f1.numerator * f2.denominator >= f2.numerator * f1.denominator;
        }

        public static Boolean operator <=(Fraction f1, Fraction f2)
        {
            return f1.numerator * f2.denominator <= f2.numerator * f1.denominator;
        }
    }

    public class Program
    {
        static Int32[] monthTable = { 0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31 };
        const Int64 MINUTE_SECONDS = 60;
        const Int64 HOUR_SECONDS = 60 * MINUTE_SECONDS;
        const Int64 DAY_SECONDS = 24 * HOUR_SECONDS;

        private static Boolean IsLeapYear(Int32 year)
        {
            return ((year % 4 == 0) && (year % 100 != 0)) || year % 400 == 0;
        }

        private static Int32 DaysInMonth(Int32 year, Int32 month)
        {
            if (IsLeapYear(year) && month == 2)
            {
                return 29;
            }

            return monthTable[month];
        }

        private static Int64 ToSeconds(CustomDateTime dateTime)
        {
            Int64 totalSeconds = 0;
            var leapYearCount = (dateTime.year - 1) / 4 - (dateTime.year - 1) / 100 +
                                (dateTime.year - 1) / 400;
            totalSeconds += ((Int64)365 * (dateTime.year - 1) + leapYearCount) * DAY_SECONDS;
            for (var month = 1; month < dateTime.month; month += 1)
            {
                totalSeconds += DaysInMonth(dateTime.year, month) * DAY_SECONDS;
            }

            totalSeconds += (dateTime.day - 1) * DAY_SECONDS;
            totalSeconds += dateTime.hour * HOUR_SECONDS;
            totalSeconds += dateTime.minute * MINUTE_SECONDS;
            totalSeconds += dateTime.second;
            return totalSeconds;
        }

        private static Fraction FractionInMonth(CustomDateTime dateTime)
        {
            Int64 totalSeconds = 0;
            totalSeconds += (dateTime.day - 1) * DAY_SECONDS;
            totalSeconds += dateTime.hour * HOUR_SECONDS;
            totalSeconds += dateTime.minute * MINUTE_SECONDS;
            totalSeconds += dateTime.second;
            return new Fraction(totalSeconds,
                DaysInMonth(dateTime.year, dateTime.month) * DAY_SECONDS);
        }

        private static Fraction FractionInYear(CustomDateTime dateTime)
        {
            Int64 totalSeconds = 0;
            for (var month = 1; month < dateTime.month; month += 1)
            {
                totalSeconds += DaysInMonth(dateTime.year, month) * DAY_SECONDS;
            }

            totalSeconds += (dateTime.day - 1) * DAY_SECONDS;
            totalSeconds += dateTime.hour * HOUR_SECONDS;
            totalSeconds += dateTime.minute * MINUTE_SECONDS;
            totalSeconds += dateTime.second;
            return new Fraction(totalSeconds, (365 + (IsLeapYear(dateTime.year)
                ? 1
                : 0)) * DAY_SECONDS);
        }

        public static void Main(String[] args)
        {
            var T = Int32.Parse(Console.ReadLine());
            for (var i = 0; i < T; i += 1)
            {
                var input = Console.ReadLine().Split(' ');
                var a1 = Int32.Parse(input[0]);
                var a2 = Int32.Parse(input[1]);
                var a3 = Int32.Parse(input[2]);
                var a4 = Int32.Parse(input[3]);
                var a5 = Int32.Parse(input[4]);
                var a6 = Int32.Parse(input[5]);
                var birthDate = new CustomDateTime(a1, a2, a3, a4, a5, a6);
                input = Console.ReadLine().Split(' ');
                var b1 = Int32.Parse(input[0]);
                var b2 = Int32.Parse(input[1]);
                var b3 = Int32.Parse(input[2]);
                var b4 = Int32.Parse(input[3]);
                var b5 = Int32.Parse(input[4]);
                var b6 = Int32.Parse(input[5]);
                var unit = Console.ReadLine();
                var currentDate = new CustomDateTime(b1, b2, b3, b4, b5, b6);

                if (unit == "Year")
                {
                    var currentFraction = FractionInYear(currentDate);
                    var birthFraction = FractionInYear(birthDate);
                    var isFullYear = currentFraction >= birthFraction
                        ? 1
                        : 0;
                    Console.WriteLine(b1 - a1 - 1 + isFullYear);
                }
                else if (unit == "Month")
                {
                    var currentFraction = FractionInMonth(currentDate);
                    var birthFraction = FractionInMonth(birthDate);
                    var isFullMonth = currentFraction >= birthFraction
                        ? 1
                        : 0;
                    Console.WriteLine((b1 - a1) * 12 + b2 - a2 - 1 + isFullMonth);
                }
                else if (unit == "Day")
                {
                    var birthTotalSeconds = ToSeconds(birthDate);
                    var currentTotalSeconds = ToSeconds(currentDate);
                    var diffSeconds = currentTotalSeconds - birthTotalSeconds;
                    Console.WriteLine(diffSeconds / DAY_SECONDS);
                }
            }
        }
    }
}